presto-orc/src/test/java/com/facebook/presto/orc/TestStructStreamReader.java

```diff
@@ -0,0 +1,293 @@
+    public void testThrowsExceptionWhenFieldNameMissing() throws IOException
+    {
+        List<String> readerFields = new ArrayList<>(Arrays.asList("field_a", "field_b", "field_c"));
+        List<String> writerFields = new ArrayList<>(Arrays.asList("field_a", "field_b", "field_c"));
+        List<String> writerData = new ArrayList<>(Arrays.asList("field_a_value",
+                                                                "field_b_value", "field_c_value"));
+        Type readerType = getTypeNullName(readerFields.size());
+        Type writerType = getType(writerFields);
+
+        write(tempFile, writerType, writerData);
+        RowBlock readBlock = read(tempFile, readerType);
+    }
+
+    /**
+     * The reader has a field that is missing from the ORC file
+     */
+    @Test
+    public void testExtraFieldsInReader() throws IOException
+    {
+        List<String> readerFields = new ArrayList<>(Arrays.asList("field_a", "field_b", "field_c"));
+
+        //field_b is missing
+        List<String> writerFields = new ArrayList<>(Arrays.asList("field_a", "field_c"));
+        List<String> writerData = new ArrayList<>(Arrays.asList("field_a_value", "field_c_value"));
+        Type readerType = getType(readerFields);
+        Type writerType = getType(writerFields);
+
+        write(tempFile, writerType, writerData);
+        RowBlock readBlock = read(tempFile, readerType);
+        List actual = (List) readerType.getObjectValue(SESSION, readBlock, 0);
+
+        assertEquals(actual.size(), readerFields.size());
+        assertEquals(actual.get(0), "field_a_value");
+        assertNull(actual.get(1));
+        assertEquals(actual.get(2), "field_c_value");
+    }
+
+    /**
+     * The ORC file has a field that is mising from t he reader
+     */
+    @Test
+    public void testExtraFieldsInWriter() throws IOException
+    {
+        //field_b is missing
+        List<String> readerFields = new ArrayList<>(Arrays.asList("field_a", "field_c"));
+        List<String> writerFields = new ArrayList<>(Arrays.asList("field_a", "field_b", "field_c"));
+        List<String> writerData = new ArrayList<>(Arrays.asList("field_a_value", "field_b_value",
+                                                                "field_c_value"));
+        Type readerType = getType(readerFields);
+        Type writerType = getType(writerFields);
+
+        write(tempFile, writerType, writerData);
+        RowBlock readBlock = read(tempFile, readerType);
+        List actual = (List) readerType.getObjectValue(SESSION, readBlock, 0);
+
+        assertEquals(actual.size(), readerFields.size());
+        assertEquals(actual.get(0), "field_a_value");
+        assertEquals(actual.get(1), "field_c_value");
+    }
+
+    private void write(TempFile tempFile, Type writerType, List<String> data) throws IOException
+    {
+        OrcWriter writer =
+                new OrcWriter(new OutputStreamOrcDataSink(new FileOutputStream(tempFile.getFile())),
```

Comment:
This is a good example of something that is better across many lines.  In cases like this, the you put the first argument on the next line.